app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt

```diff
@@ -0,0 +1,212 @@
+        imageBack = findViewById(R.id.backBtn)
+        backBtn.setOnClickListener { onBackPressed() }
+        if (intent.getBooleanExtra("isViewOrUpdate", false)) {
+            id = intent.getStringExtra("id")
+            val title = intent.getStringExtra("title")
+            val namePlace = intent.getStringExtra("namePlace")
+            val coordinate = intent.getStringExtra("coordinate")
+            val tripDate = intent.getStringExtra("tripDate")
+            val tripDateEnd = intent.getStringExtra("tripDateEnd")
+            val timeStart = intent.getStringExtra("timeStart")
+            val timeEnd = intent.getStringExtra("timeEnd")
+            image = intent.getIntExtra("image", 0)
+            val tripModel = TripModel(id, title, namePlace, coordinate, image, tripDate, tripDateEnd, timeStart, timeEnd)
+            alreadyAvailableTrip = tripModel
+            setViewOrUpdateTrip()
+        }
+        setPlace.setOnClickListener {
+            val intent = Intent(this@AddNewTrip, DetailsMap::class.java)
+            intent.putExtra("setOnlyPlace", true)
+            startActivityForResult(intent, REQUEST_CODE_ADD_NAME_PLACE)
+        }
+        setDateTimeStart.setOnClickListener{
+            isDataStart = true
+            val dialogFragment: DialogFragment = DatePicker()
+            dialogFragment.show(supportFragmentManager, "date picker")
+        }
+        setTimeStart.setOnClickListener {
+            isDataStart = false
+            val dialogFragment2: DialogFragment = DatePicker()
+            dialogFragment2.show(supportFragmentManager, "date picker")
+        }
+        saveTrip.setOnClickListener {
+            val sendTripInfoIntent = Intent()
+            val name = editTripName.text.toString().trim { it <= ' ' }
+            if (alreadyAvailableTrip != null) {
+                id = alreadyAvailableTrip!!.tripId
+                sendTripInfoIntent.putExtra("id", id)
+            }
+            sendTripInfoIntent.putExtra("nameTrip", name)
+            sendTripInfoIntent.putExtra("date", dateS) //currentDateString
+            sendTripInfoIntent.putExtra("endDate", dateE)
+            sendTripInfoIntent.putExtra("namePlace", placeNameText)
+            sendTripInfoIntent.putExtra("coordinates", coordinates)
+            sendTripInfoIntent.putExtra("timeStart", timeS)
+            sendTripInfoIntent.putExtra("timeEnd", timeE)
+            setResult(RESULT_OK, sendTripInfoIntent)
+            finish()
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setViewOrUpdateTrip() {
+        tripName!!.setText(alreadyAvailableTrip!!.title)
+        placeName!!.text = alreadyAvailableTrip!!.namePlace
+        dateOfTripStart!!.text = alreadyAvailableTrip!!.tripDate + " " + alreadyAvailableTrip!!.timeStart
+        dateS = alreadyAvailableTrip!!.tripDate
+        endDate!!.text = alreadyAvailableTrip!!.tripDateEnd + " " + alreadyAvailableTrip!!.timeEnd
```

Comment:
here you can use kotlin string interpolation: https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#concatenate-strings